Automattic/sensei
# Add quiz button block

Fixes #4406

### Changes proposed in this Pull Request

* It introduces a new Quiz button block to take the quiz through a Lesson.

### Testing instructions


* Enable the feature flag: `add_filter( 'sensei_feature_flag_course_theme', '__return_true' );`
* Create a course with a lesson and a quiz.
* Select the "Sensei Theme" as the "Course Theme" (sidebar).
* Go to the lesson in the frontend, and make sure it contains a "Take quiz" button.
* Click on the button and make sure you go to the quiz.
* Submit the quiz (to see the quiz you need to change the "Course Theme" because it's not ready yet).
* Make sure the button doesn't appear anymore (it will be accessed through another block).

--

* Also, make sure the View quiz button continues working properly on the lesson page after the refactor.


### Screenshot / Video

<img width="900" alt="Screen Shot 2021-11-23 at 17 01 51" src="https://user-images.githubusercontent.com/876340/143095438-0eb2a0a6-cf51-452a-bb2d-36eb612fab76.png">
